[image-view] Add scroll support to zoom-in and zoom-out

asiloisad opened the pull request:
It's natural to zoom-in and zoom-out images by `Ctrl-Scroll`. It was required to change how `zoom-to-fit` works.

confused-Techie commented:
Really excited about this feature idea, and something I've absent-mindedly tried to do way to many times forgetting it doesn't exist.

But I do see some test failures here, that I don't necessarily believe are the fault of your code at all, but are ones we don't expect to see, so we will probably want to take a closer look at those sometime soon.

@DeeDeeG a ping for whenever your available if you'd like to help look into this.

asiloisad commented:
My pull request introduce scroll zooming, but I think it's not best approach. I will prepare another solutions like this: https://dev.to/stackfindover/zoom-image-point-with-mouse-wheel-11n3. Please wait to check it, I will prepare a new version.

asiloisad commented:
Summary:
- zoom by `Ctrl-Scroll` added,
- `zoom-to-fit` mechanism changed, but effects remain the same
- added native background,
- added button `Center`,
- fixed bug in Pulsar of false start,
- zoom buttons target set to centre of picture (instead top-left corner),
- added `5%` scale to predefined levels,
- changed speed of navigation via arrows.

I have tested it with Pulsar and PulsarNext.

asiloisad commented:
I have tried to change the base branch of the pull request, but there are 200+ commits from the updated-latest-electron branch. How can I change the base branch to avoid this?

savetheclocktower commented:
> I have tried to change the base branch of the pull request, but there are 200+ commits from the updated-latest-electron branch. How can I change the base branch to avoid this?

I'd suggest creating a new branch off of `master`, then [cherry-picking](https://git-scm.com/docs/git-cherry-pick) the five commits in this PR onto that branch in order. Then you can open a second PR against `master`.

asiloisad commented:
I have published package [image-editor](https://packages.pulsar-edit.dev/packages/image-editor), which is superior to this changes. Let's keep image-view as simple as it is.
